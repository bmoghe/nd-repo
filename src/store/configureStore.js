import { createStore, compose, applyMiddleware } from 'redux';
import rootReducer from '../reducers';
import { createLogger } from 'redux-logger';
import promise from 'redux-promise-middleware';

const logger = createLogger({});


function configureStoreProd(initialState) {
  const middlewares = [
    // Add other middleware on this line... 
    promise,
  ];

  return createStore(rootReducer, initialState, compose(
    applyMiddleware(...middlewares)
  )
  );
}

function configureStoreDev(initialState) {
  const middlewares = [
    // Add other middleware on this line...
    // Redux middleware that spits an error on you when you try to mutate your state either inside a dispatch or between dispatches.
    // reduxImmutableStateInvariant(),
    promise,
    logger,
  ];

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

  const store = createStore(rootReducer, initialState, composeEnhancers(
    applyMiddleware(...middlewares)
  )
  );
  return store;
}

const configureStore = process.env.NODE_ENV === 'production' ? configureStoreProd : configureStoreDev;

export default configureStore;
