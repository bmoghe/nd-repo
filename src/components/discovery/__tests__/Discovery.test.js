import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { DiscoveryContainer } from '../..';

const mockStore = configureStore();
const initialState = { output: 10 };

let wrapper, store;
describe('Discovery Component', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<DiscoveryContainer store={store} docId='123' />);
  });

  it('should not crash', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
