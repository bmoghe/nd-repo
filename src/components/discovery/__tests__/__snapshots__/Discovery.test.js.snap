// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discovery Component should not crash 1`] = `ShallowWrapper {}`;
